Add new option to show or hide empty categories

mauriac opened the pull request:
<!-- Thanks for contributing to Gutenberg! Please follow the Gutenberg Contributing Guidelines:
https://github.com/WordPress/gutenberg/blob/trunk/CONTRIBUTING.md -->

## What?
<!-- In a few words, what is the PR actually doing? -->
New  option to show or hide empty categories (non-breaking change which adds functionality)

## Why?
<!-- Why is this PR necessary? What problem is it solving? Reference any existing previous issue(s) or PR(s), but please add a short summary here, too -->
Fixes #35489

## How?
<!-- How is your PR addressing the issue at hand? What are the implementation details? -->

## Testing Instructions
<!-- Please include step by step instructions on how to test this PR. -->
<!-- 1. Open a Post or Page. -->
<!-- 2. Insert a Heading Block. -->
<!-- 3. etc. -->

## Screenshots or screencast <!-- if applicable -->
<img width="640" alt="Screenshot at Mar 14 15-49-07" src="https://user-images.githubusercontent.com/41555145/158197319-23f43f44-36da-4bc2-9831-9a92eaccfde6.png">

mauriac committed:
Packages[categories-block]: Add new option to show or hide empty categories.

Fixes #35489

mauriac committed:
Update fixture for unit tests

amustaque97 commented on packages/block-library/src/categories/index.php:
```diff
@@ -22,6 +22,7 @@ function render_block_core_categories( $attributes ) {
 		'orderby'      => 'name',
 		'show_count'   => ! empty( $attributes['showPostCounts'] ),
 		'title_li'     => '',
+		'hide_empty'   => ! empty( $attributes['showEmpty'] ),
```
Thank you @mauriac  for working on this. It looks like the changes are showing inverted behaviour than expected. When I mark `Show empty categories` as checked and publish the post. It doesn't show me the empty categories. While I mark `Show empty categories` as unchecked it shows the empty categories in the published post. 

Could you please verify the behaviour?  Thank you.

mauriac replied:
Thanks for testing and raising this one. I have fixed it. feel free to test it again.

Thanks a lot.

amustaque97 replied:
Thank you @mauriac once again. Changes are working as expected. 👍🏻

mauriac committed:
Fix inverted behaviour

When "Show empty categories" is checked empty categories do not show
and vice versa. This post fix that behavious.

ntsekouras commented on packages/block-library/src/categories/block.json:
```diff
@@ -22,6 +22,10 @@
 			"default": false
+		},
+		"showEmpty": {
+		  "type": "boolean",
```
The indentation here needs to be updated(`tab`). There should be linting rule here, but it seems there were no complains 😄

mauriac replied:
hello thanks for reviewing. It's fixed

ntsekouras reviewed (commented):
Thanks for the PR @mauriac! Besides the indentation change this looks good!

mauriac committed:
Use tab for indentation

amustaque97 commented:
Thank you once again @mauriac for working on this. Let's wait for the pipeline to finish and if there are no errors. Will merge it.

mauriac commented:
@amustaque97  thanks for handling this... Please what is Gutenberg policy about branch deleting? Should I leave the branch for history or delete it to maintain the repository clean?
Thank you again.

Mamaduka commented:
@mauriac, feel free to delete the branch 👍
